### PR TITLE
typefully 1.0.9 (new cask)

### PR DESCRIPTION
**Pull request by nickpellant**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/t/typefully.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+cask "typefully" do
+  version "1.0.9"
+  sha256 :no_check
+
+  url "https://download.todesktop.com/2304250k2av6yux/Typefully%20Mac%20Installer.zip",
```
Comment (khipp) [inline Casks/t/typefully.rb:5]:
```suggestion
  arch arm: "arm64", intel: "x64"

  version "1.0.9"
  sha256 arm:   "a6d42fa9ff75bc63ee8dbad6ef02bec1290e5a9d0d184583c7ef51a3cdd512dd",
         intel: "3a09659008a97cd24d9ecd5d010ddfef343180d41743ed0d96b56f1d701d04da"

  url "https://download.todesktop.com/2304250k2av6yux/Typefully%20#{version}-#{arch}-mac.zip",
```

File: Casks/t/typefully.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  name "Typefully"
+  desc "Tool for writing and publishing tweets"
+  homepage "https://typefully.com/"
+
+  livecheck do
+    url "https://download.todesktop.com/2304250k2av6yux/latest-mac.yml"
+    strategy :electron_builder
+  end
+
+  auto_updates true
```
Comment (khipp) [inline Casks/t/typefully.rb:16]:
```suggestion
  auto_updates true
  depends_on macos: ">= :catalina"
```

File: Casks/t/typefully.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  installer script: {
+    executable: "Install Typefully.app/Contents/MacOS/ToDesktop Installer",
+  }
+
+  uninstall quit:   "com.todesktop.2304250k2av6yux",
+            delete: "/Applications/Typefully.app"
```
Comment (khipp) [inline Casks/t/typefully.rb:23]:
```suggestion
  app "Typefully.app"

  uninstall quit: "com.todesktop.2304250k2av6yux"
```

**Review by khipp — COMMENTED** [review]
Thank you for your contribution, @nickpellant! I left a few suggestions below.

**Comment by nickpellant** [comment]
Those were great suggestions, @khipp, thank you! I've updated my PR to include them.

File: Casks/t/typefully.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  end
+
+  auto_updates true
+  depends_on macos: ">= :catalina"
+
+  app "Typefully.app"
+
+  uninstall quit: "com.todesktop.2304250k2av6yux"
+
+  zap trash: "~/Library/Application Support/Typefully"
```
Comment (khipp) [inline Casks/t/typefully.rb:26]:
```suggestion
  zap trash: [
    "~/Library/Application Support/Typefully"
    "~/Library/Preferences/com.todesktop.2304250k2av6yux.plist",
    "~/Library/Logs/Typefully",
  ]
```

I found some more files to remove. Then it should be good to go.

**Comment by nickpellant** [comment]
Excellent, thanks @khipp! I've updated my PR to include your proposed changes.

To help me next time, what was your process for finding those additional files to zap?

**Comment by khipp** [comment]
@nickpellant I'm using an app uninstall tool. [Pearcleaner](https://github.com/alienator88/Pearcleaner) lists all the created files and you can copy them by using right-click.

**Comment by nickpellant** [comment]
Thank you, @khipp, that's really helpful. 🙏

**Review by khipp — APPROVED** [review]
Thank you!

**Pull request merged**